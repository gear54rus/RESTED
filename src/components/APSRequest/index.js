import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, Fields, formPropTypes } from 'redux-form';
import { Panel, Form, Col, FormGroup, Checkbox, Alert } from 'react-bootstrap';
import flow from 'lodash.flow';

import BasicAuthSection from 'components/Request/BasicAuthSection';
import * as apsActions from 'store/aps/actions';
import { getError } from 'store/aps/selectors';
import { DEFAULT_APS_REQUEST } from 'constants/constants';

import Titlebar from './Titlebar';
import URLField from './URLField';
import TokenParamsField from './TokenParamsField';
import SubmitButton from './SubmitButton';

export const apsRequestForm = 'apsRequest';

function APSRequest({ tokenFetchError, handleSubmit, sendRequest }) {
  return (
    <Panel header={<Titlebar />}>
      <Form
        horizontal
        onSubmit={handleSubmit(sendRequest)}
      >
        <FormGroup>
          <Col
            xs={5}
          >
            <Field
              name="url"
              component={URLField}
            />
          </Col>

          <Col
            xs={7}
          >
            <Fields
              names={['username', 'password']}
              component={BasicAuthSection}
              small
            />
          </Col>
        </FormGroup>
        <FormGroup>
          <Col
            xs={7}
          >
            <Field
              name="type"
              component={TokenParamsField}
            />
          </Col>

          <Col
            xs={2}
          >
            <Checkbox
              disabled // to be implemented
            >
              Refresh automatically?
            </Checkbox>
          </Col>

          <Col
            xs={3}
          >
            <SubmitButton />
          </Col>
        </FormGroup>
        {tokenFetchError && (
          <Alert bsStyle="danger">
            <strong>Failed to fetch token!</strong> {tokenFetchError.message}
          </Alert>
        )}
      </Form>
    </Panel>
  );
}

APSRequest.propTypes = {
  ...formPropTypes,
  tokenFetchError: PropTypes.instanceOf(Error),
};

const formOptions = {
  form: apsRequestForm,
};

const mapStateToProps = state => ({
  initialValues: DEFAULT_APS_REQUEST,
  tokenFetchError: getError(state),
});

export default flow(
  reduxForm(formOptions),
  connect(mapStateToProps, {
    ...apsActions,
  }),
)(APSRequest);
