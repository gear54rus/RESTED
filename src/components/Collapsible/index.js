import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Collapse, Row, Col } from 'react-bootstrap';

import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { CollapsibleButton, CollapsibleHeader, AnimatedIcon } from './StyledComponents';

export function Collapsible(props) {
  const {
    id,
    title,
    open,
    children,
    toggleCollapse,
    mountOnEnter,
    unmountOnExit,
  } = props;
  return (
    <div>
      <CollapsibleButton
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <CollapsibleHeader>
          {title}
          <AnimatedIcon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </CollapsibleHeader>
      </CollapsibleButton>
      <Collapse
        in={open}
        mountOnEnter={mountOnEnter}
        unmountOnExit={unmountOnExit}
      >
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </div>
  );
}

Collapsible.defaultProps = {
  mountOnEnter: false,
  unmountOnExit: false,
  open: false,
};

Collapsible.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  mountOnEnter: PropTypes.bool,
  unmountOnExit: PropTypes.bool,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsible);

