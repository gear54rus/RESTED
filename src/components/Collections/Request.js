import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { DragSource, DropTarget } from 'react-dnd';
import { ListGroup } from 'react-bootstrap';
import flow from 'lodash.flow';

import IconButton from 'components/IconButton';
import * as CollectionActions from 'store/collections/actions';
import * as RequestActions from 'store/request/actions';
import * as ConfigActions from 'store/config/actions';
import { isDefaultCompact } from 'store/options/selectors';
import { getEditingRequest } from 'store/config/selectors';
import selectText from 'utils/selectText';

import { StyledRequest, RequestButtons, MainContentDiv } from './StyledComponents';
import * as Type from './dropTypes';

/**
 * Specifies the drag source contract.
 */
const requestSource = {
  canDrag(props) {
    // Disallow drag if editing request
    return !props.isEditing;
  },

  beginDrag({ request, index, collectionIndex }) {
    return { request, index, collectionIndex };
  },
};

const requestTarget = {
  hover(props, monitor) {
    const dragIndex = monitor.getItem().index;
    const dragCollectionIndex = monitor.getItem().collectionIndex;
    const dragUUID = monitor.getItem().request.id;
    const hoverIndex = props.index;
    const hoverCollectionIndex = props.collectionIndex;
    const hoverUUID = props.request.id;

    // Abort further processing if dragged item is hovering over itself
    if (dragUUID === hoverUUID) {
      return;
    }

    // Note: we're mutating the monitor item here.
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    /* eslint-disable no-param-reassign */
    monitor.getItem().index = hoverIndex;
    monitor.getItem().collectionIndex = hoverCollectionIndex;
    /* eslint-enable no-param-reassign */

    // Update redux orders
    props.reorderRequest({
      // Source
      collectionIndex: dragCollectionIndex,
      requestIndex: dragIndex,
    }, {
      // Target
      collectionIndex: hoverCollectionIndex,
      requestIndex: hoverIndex,
    });
  },
};

class Request extends React.Component {
  static propTypes = {
    connectDragSource: PropTypes.func.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
    isDragging: PropTypes.bool.isRequired,
    index: PropTypes.number.isRequired,
    collectionIndex: PropTypes.number.isRequired,
    renameRequest: PropTypes.func.isRequired,
    selectRequest: PropTypes.func.isRequired,
    sendRequest: PropTypes.func.isRequired,
    deleteRequest: PropTypes.func.isRequired,
    toggleEditMode: PropTypes.func.isRequired,
    isEditing: PropTypes.bool.isRequired,
    defaultCompact: PropTypes.bool.isRequired,
    request: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string,
      method: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    }).isRequired,
  };

  state = {
    compact: this.props.defaultCompact,
  };

  toggleCompact = () => {
    this.setState({ compact: !this.state.compact });
  }

  toggleEdit = () => {
    this.props.toggleEditMode(this.props.request);
  }

  renameRequest = e => {
    const { collectionIndex, index, renameRequest } = this.props;
    e.preventDefault();

    this.toggleEdit();
    renameRequest(collectionIndex, index, this.nameRef.value);
  }

  renderRequest() {
    const { compact } = this.state;
    const { request, index, collectionIndex, isEditing } = this.props;
    const { method, name, url } = request;

    return (
      <div>
        {!compact && <h4>{method}</h4>}
        {isEditing ? (
          <form onSubmit={this.renameRequest}>
            <label
              className="sr-only"
              htmlFor={`${collectionIndex}.${index}.RequestName`}
            >
              Request name
            </label>
            <input
              autoFocus
              onFocus={selectText}
              id={`${collectionIndex}.${index}.RequestName`}
              defaultValue={name}
              ref={ref => { this.nameRef = ref; }}
            />
            <IconButton
              tooltip="Save"
              icon="check"
            />
          </form>
        ) : (
          name || url
        )}
      </div>
    );
  }

  render() {
    const { compact } = this.state;
    const {
      connectDragSource,
      connectDropTarget,
      isDragging,
      request,
      collectionIndex,
      selectRequest,
      sendRequest,
      deleteRequest,
      isEditing,
    } = this.props;

    return connectDragSource(connectDropTarget(
      <div>{/* Need a wrapper div for React DnD support */}
        <StyledRequest isDragging={isDragging}>
          <ListGroup componentClass="div">
            <div className="list-group-item">
              <RequestButtons compact={compact}>
                {!compact && (
                  <IconButton
                    tooltip="Toggle edit"
                    icon="pencil-alt"
                    onClick={this.toggleEdit}
                  />
                )}
                {compact && (
                  <IconButton
                    tooltip="Expand"
                    icon="plus"
                    onClick={this.toggleCompact}
                  />
                )}

                {!compact && !isEditing && (
                  <IconButton
                    tooltip="Minimize"
                    icon="minus"
                    onClick={this.toggleCompact}
                  />
                )}
                {!compact && isEditing && (
                  <IconButton
                    tooltip="Delete"
                    icon="trash"
                    onClick={() => deleteRequest(request.id, collectionIndex)}
                  />
                )}
              </RequestButtons>

              {isEditing ? (
                <MainContentDiv compact={compact}>
                  {this.renderRequest()}
                </MainContentDiv>
              ) : (
                <MainContentDiv
                  compact={compact}
                  onClick={() => selectRequest(request)}
                  onDoubleClick={() => sendRequest(request)}
                >
                  {this.renderRequest()}
                </MainContentDiv>
              )}
            </div>
          </ListGroup>
        </StyledRequest>
      </div>,
    ));
  }
}

const mapStateToProps = (state, props) => ({
  isEditing: getEditingRequest(state)
    ? getEditingRequest(state).id === props.request.id
    : false,
  defaultCompact: isDefaultCompact(state),
});

export default flow(
  DropTarget(Type.Request, requestTarget, connector => ({
    connectDropTarget: connector.dropTarget(),
  })),
  DragSource(Type.Request, requestSource, (connector, monitor) => ({
    connectDragSource: connector.dragSource(),
    isDragging: monitor.isDragging(),
  })),
  connect(mapStateToProps, {
    ...CollectionActions,
    ...RequestActions,
    ...ConfigActions,
  }),
)(Request);

