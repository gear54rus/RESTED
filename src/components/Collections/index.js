import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import IconButton from 'components/IconButton';
import * as Actions from 'store/collections/actions';

import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix">
      <IconButton
        onClick={() => addCollection()}
        tooltip="Add new collection"
        icon="plus"
        className="pull-right"
      />
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel>
      <Panel.Heading>
        <Panel.Title
          componentClass={Titlebar}
          addCollection={addCollection}
        />
      </Panel.Heading>
      <Panel.Body>
        <CollectionList />
      </Panel.Body>
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

