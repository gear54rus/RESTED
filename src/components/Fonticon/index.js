import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Fonticon({ icon, className }) {
  return (
    <i
      className={classNames('fas', `fa-${icon}`, className)}
      role="presentation"
    />
  );
}

Fonticon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Fonticon;
