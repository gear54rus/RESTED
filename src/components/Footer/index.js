import React from 'react';

import { StyledFooter } from './StyledComponents';

export default function Footer() {
  return (
    <StyledFooter>
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub
        </a>
      </small>
      ·
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS/issues"
          target="_blank"
          rel="noopener noreferrer"
        >
          Report a bug
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gitter.im/RESTEDclient"
          target="_blank"
          rel="noopener noreferrer"
        >
          Chat (RESTED)
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gist.github.com/esphen/f50cc049c8ed1256a11763826dac97db"
          target="_blank"
          rel="noopener noreferrer"
        >
          Licenses (RESTED)
        </a>
      </small>
    </StyledFooter>
  );
}
