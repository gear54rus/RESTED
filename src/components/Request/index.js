import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { reduxForm, Fields, FieldArray, FormSection, getFormValues } from 'redux-form';
import { Panel, Form } from 'react-bootstrap';
import flow from 'lodash.flow';

import * as requestActions from 'store/request/actions';
import * as collectionsActions from 'store/collections/actions';
import { isEditMode } from 'store/config/selectors';
import { DEFAULT_REQUEST } from 'constants/constants';
import AuthField from 'components/Authentication';

import Titlebar from './Titlebar';
import LineFields from './LineFields';
import HeadersField from './HeadersField';
import BodyField from './BodyField';

export const requestForm = 'request';

function Request(props) {
  const {
    formValues = {},
    handleSubmit,
    editMode,
    sendRequest,
    updateRequest,
  } = props;

  return (
    <Panel>
      <Panel.Heading>
        <Panel.Title componentClass={Titlebar} />
      </Panel.Heading>
      <Panel.Body>
        <Form
          horizontal
          onSubmit={handleSubmit(editMode ? updateRequest : sendRequest)}
        >
          <Fields
            names={['method', 'url']}
            component={LineFields}
          />
          <FieldArray
            name="headers"
            component={HeadersField}
          />
          <FormSection
            name="auth"
            component={AuthField}
          />
          {!['GET', 'HEAD'].includes(formValues.method) && (
            <BodyField />
          )}
        </Form>
      </Panel.Body>
    </Panel>
  );
}

Request.propTypes = {
  formValues: PropTypes.shape({}),
  handleSubmit: PropTypes.func.isRequired,
  editMode: PropTypes.bool.isRequired,
  sendRequest: PropTypes.func.isRequired,
  updateRequest: PropTypes.func.isRequired,
};

const formOptions = {
  form: requestForm,
};

const mapStateToProps = state => ({
  useFormData: state.request.useFormData,
  initialValues: DEFAULT_REQUEST,
  formValues: getFormValues(requestForm)(state),
  editMode: isEditMode(state),
});

export { Request };
export default flow(
  reduxForm(formOptions),
  connect(mapStateToProps, {
    ...requestActions,
    ...collectionsActions,
  }),
)(Request);
