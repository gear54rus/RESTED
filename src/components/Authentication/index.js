import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { FormSection, Field } from 'redux-form';
import { FormGroup, FormControl, Col, ControlLabel } from 'react-bootstrap';

import { textFieldShape } from 'propTypes/field';
import Collapsible from 'components/Collapsible';
import { authTypes } from 'store/auth/sagas';
import { getAuthType } from 'store/request/selectors';

import BasicFields from './BasicAuthFields';
import APSTokenFields from './APSTokenFields';
import OAuth1Fields from './OAuth1Fields';

function AuthTypeField({ input }) {
  return (
    <FormGroup controlId="auth">
      <Col
        componentClass={ControlLabel}
        xs={2}
      >
        Type
      </Col>
      <Col xs={8}>
        <FormControl
          componentClass="select"
          placeholder="Authentication type"
          {...input}
        >
          {Object.entries(authTypes).map(([type, { caption }]) => (
            <option
              key={type}
              value={type}
            >
              {caption}
            </option>
          ))}
        </FormControl>
      </Col>
    </FormGroup>
  );
}

AuthTypeField.propTypes = textFieldShape;

function Authentication({ authType }) {
  let authFields;

  switch (authType) {
    case 'basic':
      authFields = BasicFields;
      break;
    case 'apsToken':
      authFields = APSTokenFields;
      break;
    case 'oauth1':
      authFields = OAuth1Fields;
      break;
    default:
      authFields = null;
  }

  return (
    <Collapsible
      title="Authentication"
      id="auth"
    >
      <Field
        name="type"
        component={AuthTypeField}
      />
      {authFields && (
        <FormSection
          name={authType}
          component={authFields}
        />
      )}
    </Collapsible>
  );
}

Authentication.propTypes = {
  authType: PropTypes.oneOf(Object.keys(authTypes)),
};

const mapStateToProps = state => ({
  authType: getAuthType(state),
});

export default connect(mapStateToProps)(Authentication);
