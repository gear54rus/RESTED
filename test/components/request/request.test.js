import React from 'react';
import { Provider } from 'react-redux';
import { reduxForm } from 'redux-form';
import renderer from 'react-test-renderer';

/* eslint-disable import/no-unresolved */
import { Request } from 'components/Request';

import makeStore from '../../makeStore';

it('should render correctly', () => {
  const props = {
    updateRequest() {},
    sendRequest() {},
    editMode: false,
    collectionsMinimized: false,
  };
  const onSubmitFail = jest.fn();
  const store = makeStore({
    form: {
      request: {
        values: {
          bodyType: 'json',
          auth: { type: 'disabled' },
        },
      },
    },
  });

  const Decorated = reduxForm({
    form: 'request',
    onSubmitFail,
  })(Request);

  const tree = renderer.create(
    <Provider store={store}>
      <Decorated {...props} />
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
  expect(onSubmitFail).not.toHaveBeenCalled();
});

