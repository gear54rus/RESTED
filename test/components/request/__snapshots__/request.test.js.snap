// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <div
      className="sc-gZMcBi kWnQrH clearfix"
    >
      <h2
        className="pull-left"
      >
        Request
      </h2>
      <button
        className="pull-right hidden-xs sc-bZQynM jYuQaZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-gzVnrw jkZMEg"
        >
          <div
            className="sc-htoDjs iTnqOg"
          />
          <span
            className="sc-dnqmqq jGQnGF"
          >
            Add to collection
          </span>
        </div>
        <i
          className="fas fa-plus"
          role="presentation"
        />
      </button>
      <button
        className="pull-right sc-bZQynM jYuQaZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-gzVnrw jkZMEg"
        >
          <div
            className="sc-htoDjs iTnqOg"
          />
          <span
            className="sc-dnqmqq jGQnGF"
          >
            Options
          </span>
        </div>
        <i
          className="fas fa-cogs"
          role="presentation"
        />
      </button>
      <button
        className="pull-right hidden-xs sc-bZQynM jYuQaZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-gzVnrw jkZMEg"
        >
          <div
            className="sc-htoDjs iTnqOg"
          />
          <span
            className="sc-dnqmqq jGQnGF"
          >
            Hide collections
          </span>
        </div>
        <i
          className="fas fa-expand"
          role="presentation"
        />
      </button>
    </div>
  </div>
  <div
    className="panel-body"
  >
    <form
      className="form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="url"
        >
          URL
        </label>
        <div
          className="col-sm-10"
        >
          <input
            autoFocus={true}
            className="form-control"
            id="url"
            name="url"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="method"
        >
          Method
        </label>
        <div
          className="col-sm-7"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-6"
            >
              <select
                className="form-control"
                id="method"
                name="method"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Method"
                type={undefined}
                value="CUSTOM"
              >
                <option
                  value="GET"
                >
                  GET
                </option>
                <option
                  value="POST"
                >
                  POST
                </option>
                <option
                  value="PUT"
                >
                  PUT
                </option>
                <option
                  value="DELETE"
                >
                  DELETE
                </option>
                <option
                  value="PATCH"
                >
                  PATCH
                </option>
                <option
                  value="HEAD"
                >
                  HEAD
                </option>
                <option
                  value="JSONP"
                >
                  JSONP
                </option>
                <option
                  value="CUSTOM"
                >
                  CUSTOM
                </option>
              </select>
            </div>
            <div
              className="col-xs-6"
            >
              <input
                className="form-control"
                id="method"
                name="method"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type={undefined}
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Send request
          </button>
        </div>
      </div>
      <div>
        <button
          className="sc-htpNat iPErUs btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <h4
            className="sc-bdVaJa bcSwbK"
          >
            Headers
            <i
              className="fas fa-angle-right sc-bwzfXH dSpMtd"
              role="presentation"
            />
          </h4>
        </button>
        <div
          aria-expanded={null}
          className="collapse row"
        >
          <div
            className="col-xs-12"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <button
                  className="sc-gqjmRU ivCeJB btn btn-link"
                  disabled={false}
                  id="addHeaderButton"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="fas fa-plus"
                    role="presentation"
                  />
                  Add header
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <button
          className="sc-htpNat iPErUs btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <h4
            className="sc-bdVaJa bcSwbK"
          >
            Authentication
            <i
              className="fas fa-angle-right sc-bwzfXH dSpMtd"
              role="presentation"
            />
          </h4>
        </button>
        <div
          aria-expanded={null}
          className="collapse row"
        >
          <div
            className="col-xs-12"
          >
            <div
              className="form-group"
            >
              <label
                className="col-xs-2 control-label"
                htmlFor="auth"
              >
                Type
              </label>
              <div
                className="col-xs-8"
              >
                <select
                  className="form-control"
                  id="auth"
                  name="auth.type"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Authentication type"
                  type={undefined}
                  value="disabled"
                >
                  <option
                    value="disabled"
                  >
                    Disabled
                  </option>
                  <option
                    value="basic"
                  >
                    Basic
                  </option>
                  <option
                    value="apsToken"
                  >
                    APS Token
                  </option>
                  <option
                    value="oauth1"
                  >
                    OAuth 1.0a
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <button
          className="sc-htpNat iPErUs btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <h4
            className="sc-bdVaJa bcSwbK"
          >
            Request body
            <i
              className="fas fa-angle-right sc-bwzfXH dSpMtd"
              role="presentation"
            />
          </h4>
        </button>
        <div
          aria-expanded={null}
          className="collapse row"
        >
          <div
            className="col-xs-12"
          >
            <div
              className="form-group"
            >
              <label
                className="col-xs-2 control-label"
                htmlFor="bodyType"
              >
                Type
              </label>
              <div
                className="col-xs-8"
              >
                <select
                  className="form-control"
                  id="bodyType"
                  name="bodyType"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Body type"
                  type={undefined}
                  value="json"
                >
                  <option
                    value="custom"
                  >
                    Custom
                  </option>
                  <option
                    value="json"
                  >
                    JSON
                  </option>
                  <option
                    value="multipart"
                  >
                    Multipart form data
                  </option>
                  <option
                    value="urlencoded"
                  >
                    URLencoded form data
                  </option>
                </select>
              </div>
            </div>
            <div
              className="sc-jTzLTM gveISI"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-12"
                >
                  <button
                    className="sc-gqjmRU ivCeJB btn btn-link"
                    disabled={false}
                    id="addParameter"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fas fa-plus"
                      role="presentation"
                    />
                    Add parameter
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
